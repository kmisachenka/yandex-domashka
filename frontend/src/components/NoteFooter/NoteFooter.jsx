import React from 'react';
import * as PropTypes from 'prop-types';
import moment from 'moment';

import styles from './NoteFooter.module.scss';

import { ReactComponent as DoneButton } from './svg/done.svg';
import { ReactComponent as EditButton } from './svg/edit.svg';

import { Tags } from '../../containers';
import { Note } from '../../types';

moment.locale('ru');

export default function NoteFooter(props) {
  const { note, tags, handleDoneClick } = props;
  const { id, created, archived } = note;
  return (
    <div className={styles.footer}>
      {tags && <Tags tags={tags} />}
      <div className={styles['last-line']}>
        <ul className={styles.buttons}>
          <li className={styles.item}>
            <button
              type="button"
              className={styles.button}
              disabled={archived}
              onClick={e => {
                e.preventDefault();
                handleDoneClick(id);
              }}
            >
              <DoneButton />
            </button>
          </li>
          <li className={styles.item}>
            <button type="button" className={styles.button} disabled={archived}>
              <EditButton />
            </button>
          </li>
        </ul>
        <span className={styles.time}>{moment(created).fromNow()}</span>
      </div>
    </div>
  );
}

NoteFooter.defaultProps = {
  tags: [],
};

NoteFooter.propTypes = {
  note: PropTypes.shape(Note).isRequired,
  tags: PropTypes.arrayOf(PropTypes.number.isRequired),
  handleDoneClick: PropTypes.func.isRequired,
};
