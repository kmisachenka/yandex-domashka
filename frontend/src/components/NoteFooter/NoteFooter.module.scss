.footer {
  display: flex;
  flex-direction: column;

  justify-content: space-between;
  padding-bottom: 24px;

  padding-left: 24px;
  padding-right: 24px;
}

.buttons {
  color: transparent;

  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.button {
  appearance: none;

  background: none;
  border: 0;
  margin: 0;

  opacity: 0.3;

  padding: 5px;

  &:focus {
    opacity: 1;
  }

  &:active {
    background: rgba(51, 51, 51, 0.3);
    border-radius: 4px;
    opacity: 1;
  }

  &:hover {
    opacity: 1;
  }
}

.item {
  margin-right: 5px;

  &:last-child {
    margin-right: 0;
  }
}

.time {
  font-family: Helvetica, serif;
  font-size: 16px;
  line-height: 20px;
  text-align: right;
}

.last-line {
  align-items: center;
  display: flex;
  justify-content: space-between;
}
