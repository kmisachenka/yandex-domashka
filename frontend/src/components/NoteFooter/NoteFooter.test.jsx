import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';

import NoteFooter from './NoteFooter';

jest.mock('moment');
moment.mockReturnValue({
  fromNow: () => '2 дня',
});

const note = {
  id: 3,
  type: 'text',
  text: 'Купить чайник с Bluetooth\nВидел у Xiaomi на днях',
  tags: [0],
  size: 's',
  created: 1551593220000,
};
const handleDoneClick = jest.fn();
describe('NoteFooter', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <NoteFooter note={note} handleDoneClick={handleDoneClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
