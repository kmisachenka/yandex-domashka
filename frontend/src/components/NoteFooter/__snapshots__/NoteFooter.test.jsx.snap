// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteFooter should match snapshot 1`] = `
<div
  className="footer"
>
  <Connect(TagsContainer)
    tags={Array []}
  />
  <div
    className="last-line"
  >
    <ul
      className="buttons"
    >
      <li
        className="item"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          <ForwardRef />
        </button>
      </li>
      <li
        className="item"
      >
        <button
          className="button"
          type="button"
        >
          <ForwardRef />
        </button>
      </li>
    </ul>
    <span
      className="time"
    >
      2 дня
    </span>
  </div>
</div>
`;
